### PR TITLE
refactor: move fr32 into its own crate

### PR DESCRIPTION
**Pull request by DrPeterVanNostrand**
Moves `storage-proofs::fr32` into it's own crate. 

Issue #1343

### CODE REVIEW

**Comment by dignifiedquire** [comment]
There are more things in filecoin-proofs `fr32.rs` and `fr32_reader.rs`, can you pull those in too please?

**Comment by DrPeterVanNostrand** [comment]
@dignifiedquire yup, np

File: fr32/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+bitvec = "0.17"
+criterion = "0.3"
+itertools = "0.9"
+pretty_assertions = "0.6.1"
+rand = "0.7"
+rand_xorshift = "0.2.0"
+
+[features]
+default = ["pairing"]
+pairing = ["bellperson/pairing"]
```
Comment (dignifiedquire) [inline fr32/Cargo.toml:28]: needs a `gpu` feature, enabling `bellperson/gpu`

**Review by dignifiedquire — CHANGES_REQUESTED** [review]
please enable ci for the new crate

**Comment by cryptonemo** [comment]
@DrPeterVanNostrand When you rebase and push, the CI should complete as expected (at least the phase2 related jobs that don't exist any longer won't be required)

File: storage-proofs/post/Cargo.toml [hunk 2]
```diff
@@ -38,5 +39,5 @@ filecoin-hashers = { path = "../../filecoin-hashers", version = "1.0.0", default
 [features]
 default = ["pairing", "gpu"]
 gpu = ["storage-proofs-core/gpu", "filecoin-hashers/gpu"]
```
Comment (dignifiedquire) [inline storage-proofs/post/Cargo.toml:41]: this and the other crates depending on `fr32` will now need to enable the `gpu` feature

**Review by dignifiedquire — CHANGES_REQUESTED** [review]
almost, just some cargo feature updates missing

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**